contracts/Tokenomics.sol

```diff
@@ -912,8 +944,12 @@ contract Tokenomics is TokenomicsConstants {
                 }
                 // Record amounts data only if at least one incentive unit fraction is not zero
                 if (incentiveFlags[unitType] || incentiveFlags[unitType + 2]) {
-                    // The amount has to be adjusted for the number of units in the service
+                    // Amount and voting power has to be adjusted for the number of units in the service
                     uint96 amount = uint96(amounts[i] / numServiceUnits);
+                    uint96 vPowerUnit;
+                    if (vPower > 0) {
+                        vPowerUnit = uint96(vPower / numServiceUnits);
```

Comment:
veOLAS power is then further distributed between service units. Note that the fraction for component / agent normalizer will be applied later in the finalization of top-ups.